### ISSUE TITLE
Fix S3218 FN: Support static abstract/virtual methods in interfaces

### ISSUE THREAD

**Issue by cristian-ambrosini-sonarsource** [SonarSource/sonar-dotnet#6388]
### Description

FN for S3218 (MemberShadowsOuterStaticMember).

Using C#11 static abstract/virtual methods in interfaces, the rule is not raising an issue.

### Repro steps

```csharp
            public interface IMath<T>
{
    public static abstract T AbstractMethod(T t);

    public static virtual T VirtualMethod(T t)
    {
        return t;
    }

    public class Inner
    {
        public T AbstractMethod(T t) // Compliant, FN
        {
            return t;
        }

        public T VirtualMethod(T t) // Compliant, FN
        {
            return t;
        }
    }
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
